lib/src/themes/expression/parser/expression_parser.dart

```diff
@@ -96,6 +96,13 @@ class ExpressionParser {
       }
     }
     if (json is Map) {
+      if (json['type'] == 'categorical') {
```

Comment:
Thanks for the improvement! It would be good to have a test for this, you can find examples in `test/src/themes/expression/expression_parser_test.dart`

Reply:
Thanks for the review @greensopinion and for hinting to the tests. If I should be honest, I was still a bit overwhelmed by the mapbox style specs and the package structure when creating this pull request. :D 
I'll try to add a test and additional type safetly.